swan-lake/learn-the-language/distinctive-language-features/advanced-general-purpose-language-features.md

```diff
@@ -745,20 +745,20 @@ Two constructs are utilized to provide the output of these lang library function
 import ballerina/io;
 import ballerina/lang.regexp;
 
-string[] names = re `,`.split("Bob,Frank,Will,Jack"); // ["Bob","Frank","Will","Jack"]
-
-int patternCount = re `[bB].tt[a-z]*`.findAll("Butter was bought by Betty but the butter was bitter.").length(); // 4
+public function main() {
+    string[] names = re `,`.split("Bob,Frank,Will,Jack"); // ["Bob","Frank","Will","Jack"]
+    
+    int patternCount = re `[bB].tt[a-z]*`.findAll("Butter was bought by Betty but the butter was bitter.").length(); // 4
 
-string result = re `0+`.replaceAll("10010011", "*"); // 1*1*11
+    string result = re `0+`.replaceAll("10010011", "*"); // 1*1*11
 
-// Extract the username and domain name from an email
-regexp:Groups? emailGroups = re `([a-z]+)@([a-z]+\.[a-z]{2,})`.findGroups("bob@example.net");
-if emailGroups is regexp:Groups {
-    regexp:Span username = <regexp:Span>emailGroups[1];
-    // Prints the username, start index and end index of the username substring
-    io:println(username.substring());
-    io:println(username.startIndex);
-    io:println(username.endIndex);
+    // Extract the username and domain name from an email
+    regexp:Groups? emailGroups = re `([a-z]+)@([a-z]+\.[a-z]{2,})`.findGroups("bob@example.net");
+    if emailGroups is regexp:Groups {
+        regexp:Span username = <regexp:Span>emailGroups[1];
+        // Prints the matched substring, start index, and end index
+        io:println(string `substring: ${username.substring()} start: ${username.startIndex} end: ${username.endIndex}`);
+    }
```

Comment:
Wouldn't it be better to specify that the domain name part would be in the 3rd group since it is not being printed here?
```
// <regexp:Span>emailGroups[1] - username
// <regexp:Span>emailGroups[2] - domain name
```